src/base/preferences.cpp

```diff
@@ -112,6 +110,9 @@ QString Preferences::getLocale() const
 
 void Preferences::setLocale(const QString &locale)
 {
+    if (locale == getLocale())
```

Comment:
I wonder if this could be done at some lower level (e.g. in `setValue()`) instead of repeating such checking in all the setters.

Reply:
I already thought about it but all of it will move more code and require more code when adding a new setting.

Reply:
Well, it's is due to default value handling, IIRC.

Reply:
BTW, I had an idea of using SettingValue (probably, changed a bit) in Preferences.

Reply:
>BTW, I had an idea of using SettingValue (probably, changed a bit) in Preferences.

Won't that create more boilerplate than it reduce? Anyway I'm not too interested in it.